### PR TITLE
Feature/bib 522 fix code todos 09 2020

### PR DESCRIPTION
**Pull request by sinejespersen**
https://jira.itkdev.dk/browse/BIB-522

### CODE REVIEW

File: assets/js/App.js [hunk 1]
```diff
@@ -5,151 +5,116 @@
...
         });
 
+        // Configuration recieved from backend.
         socket.on('Configuration', (data) => {
             setBoxConfig(data);
+            createIdleTimer(data.inactivityTimeOut);
         });
 
         // Listen for changes to machine state.
         socket.on('UpdateState', (data) => {
```
Comment (sinejespersen) [inline assets/js/App.js:65]: The idletimer is initiated when we have the configuration

File: assets/js/Steps/Bibbox.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
```
Comment (sinejespersen) [inline assets/js/Steps/Bibbox.js:1]: old app.js

File: assets/js/Steps/Bibbox.spec.js [hunk 3]
```diff
@@ -0,0 +1,1062 @@
+/**
```
Comment (sinejespersen) [inline assets/js/Steps/Bibbox.spec.js:1]: Most of the tests are located in the bibbox.spec. This is due to the way the state is implemented, and the components that depends on the machinestate cannot be tested without the machinestate, so they have to be tested through the Bibbox component

File: assets/js/Steps/CheckInItems.js [hunk 4]
```diff
@@ -31,74 +33,75 @@ import Input from './components/input';
...
-            actionHandler('checkInItem', {
-                itemIdentifier: code
-            });
         };
 
         barcodeScanner.start(barcodeCallback);
         return () => {
             barcodeScanner.stop();
         };
     }, [actionHandler]);
```
Comment (sinejespersen) [inline assets/js/Steps/CheckInItems.js:75]: It now checks for the items (at some point the state machine send [] and not nothing, I think?)

**Review by tuj — APPROVED** [review]

**Review by cableman — APPROVED** [review]
But not fan of the `eslint-disable-line new-cap` at line endings, but I will not stop progress do to them

**Pull request merged**